tensorflow/python/ops/signal/fft_ops.py

```diff
@@ -358,11 +358,11 @@ def fftshift(x, axes=None, name=None):
     x = _ops.convert_to_tensor(x)
     if axes is None:
       axes = tuple(range(x.shape.ndims))
-      shift = [int(dim // 2) for dim in x.shape]
+      shift = [dim // 2 for dim in _array_ops.shape(x)]
```

Comment:
Tensors are only iterable in eager mode, how about:
```
shift = _array_ops.shape(x) // 2
```

Reply:
yes didn't catch that !